man/notify-selfcontained-example.c

```diff
@@ -0,0 +1,167 @@
+/* SPDX-License-Identifier: MIT-0 */
+
+/* Implement the systemd notify protocol without external dependencies.
+ * Supports both readiness notification on startup and on reloading,
+ * according to the protocol defined at:
+ * https://www.freedesktop.org/software/systemd/man/latest/sd_notify.html
+ * This protocol is guaranteed to be stable as per:
+ * https://systemd.io/PORTABILITY_AND_STABILITY/ */
+
+#include <assert.h>
+#include <errno.h>
+#include <inttypes.h>
+#include <signal.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <time.h>
+#include <unistd.h>
+
+#define _cleanup_(f) __attribute__((cleanup(f)))
+
+bool reloading = 0;
+bool terminating = 0;
+
+static void closep(int *fd) {
+  if (!fd || *fd < 0)
+    return;
+
+  close(*fd);
+  *fd = -1;
```

Comment:
-EBADF?

Reply:
we stole the concept of -EBADF from LXC, so there are now at least two projects which use -EBADF instead of -1

Reply:
I think, any negative integer counts as an invalid fd, but -1 is the more traditional value as it is returned by `open` in case of error.

Reply:
> Hmm, since this is an example for other projects, I think the convention was still `-1`?

Indeed that's the reason, this is public documentation and clarity is more important, I don't want readers to think there's a special value associated with it, given it's just a code style convention